### ISSUE TITLE
Bug: If I upload one model and then upload a second model, the page will move forward after the first model has finished uploading, resulting in a blank page.

### ISSUE THREAD

**Issue by Dansken** [all3dp/printing-engine-client#1140]
How to reproduce: 

- Upload a model with a large file size. 
- While the upload is underway, begin the process of uploading a second model.
- Wait until the first model has been uploaded.

![image](https://user-images.githubusercontent.com/25741371/59493064-ae280480-8e8a-11e9-9519-5d5f962f7501.png)

**Comment by Dansken** [comment]
@ranzwertig @webholics 

I have just tested this feature and while the issue is fixed, I have a new issue that have been arisen. 

If I upload one model and start the upload of a second larger file, clicking the "Configure Selection" button will take me to the material selection page. Once the upload of the second model is done, the model is not added to the model list on the material page but it is included in the selected models. 

How to reproduce: 
- Upload one model
- Upload a second model (should be larger to make sure the next step can be done).
- While the second model is uploading, click on "Configure Selection"
- See list of models (1 of 1)
- When second model is done uploading, list now says (2 of 2) but the list only includes the first model.

![image](https://user-images.githubusercontent.com/25741371/59510991-6a96c000-8eb5-11e9-9c7c-fbae71bfcb97.png)
 
![image](https://user-images.githubusercontent.com/25741371/59511009-74202800-8eb5-11e9-9234-9401905f8cde.png)
